### ISSUE TITLE
[a11y-defect-3]: Payment History table extends past right edge of viewport at 200-250% browser zoom

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#95134]
### Point of contact

Trevor Pierce

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The Your VA Payments view has a table of payments that starts to extend outside the right edge of the viewport when I zoom my browser to 200-250%. This could prevent some users from seeing all of the data without zooming in to a higher resolution where the table adjusts to our responsive layout. Screenshot attached below.

There is a possibility to zoom in farther so the table switches to its mobile-first view, but (for now) the table headers are hidden, thus trading one affordance for another.

---

![Image](https://github.com/user-attachments/assets/438a1619-9c99-4aa9-b4a9-7bcef5a9e2b1)

### Reproduction steps

1. Log in with a test user who has a payment history with VA
2. Navigate to va-payment-history/payments/
3. Set your browser width to 1280px and zoom in to at least 200%, or 250%
4. Verify the table's right edge is out of the view port and cannot be scrolled horizontally

### Proposed solution or next steps

I think there could be a few options here:

1. Adding an optional prop to the VA-table component that allows it to take a `tabindex="0"` so the table can be focused and scrolled using a keyboard
2. Add a container with the tabindex on this particular view
3. Adjust the CSS to switch to the responsive layout sooner

### References, articles, or WCAG support

1. Understanding [SC 1.4.10](https://w3.org/TR/WCAG22#reflow): Reflow (Level AA)

### Type of issue

- [ ] Axe-core
- [ ] Screenreader | assistive tech & device support
- [ ] Keyboard
- [x] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by dysbo** [comment]
Possible question for design system team - show them the issue and propose the solution, and make sure this is not just a "we're doing it wrong" type of issue

**Comment by 1Copenut** [comment]
> I tested locally using Storybook and I've been able to scroll horizontally to access all of the information (using mouse and keyboard right arrow) when the table is zoomed.

I appreciate the question @jamigibbs. Chrome on Windows and Chrome and Safari on MacOS do not automatically set focus on tables that overflow the horizontal viewport. I cannot use the arrow keys to scroll horizontally unless we add that `tabindex="0"` affordance to the containing div.

**Comment by jamigibbs** [comment]
> Chrome on Windows and Chrome and Safari on MacOS do not automatically set focus on tables that overflow the horizontal viewport. I cannot use the arrow keys to scroll horizontally unless we add that tabindex="0" affordance to the containing div.

> Adding an optional prop to the VA-table component that allows it to take a tabindex="0" so the table can be focused and scrolled using a keyboard

Thanks @1Copenut! I'm curious then if adding `tabindex` needs to be optional? Is the intention to only have that set if the table has enough content to make it overflow? Are there any cons to simply having it there in all scenarios?

**Comment by 1Copenut** [comment]
Yes, our (BMT II) intent was to add an optional prop to give us a way to make content scrollable if it overflowed the viewport, but avoid adding another tabstop to the DOM if content wasn't overflowing. The only con, if we can call it that is the extra tabstop for keyboard and possibly screen reader users.

**Comment by jamigibbs** [comment]
@1Copenut Thanks again. I'm digging into this a little more. The table web component is using the [USWDS table](https://designsystem.digital.gov/components/table/) markup and style. I see now their guidance for the [scrollable table variation](https://designsystem.digital.gov/components/table/#scrollable-table) says:

> **Scrollable tables need to be focusable.** When you use the `.usa-table-container--scrollable` variant with a table, you must add the `tabindex="0"` attribute to the scrollable element. This attribute assures that users navigating with a keyboard are able to select and scroll the table. `tabindex="0"` enables focus on elements that do not get focus by default. This attribute does not change the tab order. It places the element in the logical navigation flow.

We actually have adding the [scrollable variation for USWDS table in our current sprint](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3448). That means we should probably be creating a `scrollable` prop which will add the `.usa-table-container--scrollable` class and `tabindex` assignment (just like USWDS outlines above).

@janechodance Since you've already made good progress [on your PR](https://github.com/department-of-veterans-affairs/component-library/pull/1428), what are your thoughts on updating it so that your prop is named "scrollable" instead of "isFocusable". You would also conditionally add the `.usa-table-container--scrollable` class to the same element that has the `tabindex` (it should already be getting imported from the USWDS module). I think that would bring us in line with USWDS and still solve your defect.

**Comment by jamigibbs** [comment]
> [@jamigibbs](https://github.com/jamigibbs), thanks for digging into this. I can definitely make the change. However, I'm worried "scrollable" may be confusing as the table is scrollable just not for keyboard user. So maybe use "keyboardScrollable"? Let me know what you think. Thank you!

@janechodance  yeah I see your point. I think though the reason that it's already scrolling for the mouse is because of [this recent update](https://github.com/department-of-veterans-affairs/component-library/pull/1416/commits/a58d8c7061e5de2fbc56bd3f8c5e43365179b5a2) which we might also want to revert if the `.usa-table-container--scrollable` and `tabindex` does the job for both keyboard and mouse. 

If that's the case, I think naming the prop `scrollable` would then make sense because it's covering both keyboard and mouse. I can do some testing to confirm though.

**Comment by dysbo** [comment]
Design System change has been pushed, waiting for that to become available for `vets-website` and then can move forward
